src/Datadog.Trace/Logging/DatadogSerilogLogger.cs

```diff
@@ -0,0 +1,210 @@
+            => Write(LogEventLevel.Error, exception, messageTemplate, args, sourceLine, sourceFile);
+
+        private void Write<T>(LogEventLevel level, Exception exception, string messageTemplate, T property, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid boxing + array allocation if disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, new object[] { property }, sourceLine, sourceFile);
+            }
+        }
+
+        private void Write<T0, T1>(LogEventLevel level, Exception exception, string messageTemplate, T0 property0, T1 property1, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid boxing + array allocation if disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, new object[] { property0, property1 }, sourceLine, sourceFile);
+            }
+        }
+
+        private void Write<T0, T1, T2>(LogEventLevel level, Exception exception, string messageTemplate, T0 property0, T1 property1, T2 property2, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid boxing + array allocation if disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, new object[] { property0, property1, property2 }, sourceLine, sourceFile);
+            }
+        }
+
+        private void Write(LogEventLevel level, Exception exception, string messageTemplate, object[] args, int sourceLine, string sourceFile)
+        {
+            if (_logger.IsEnabled(level))
+            {
+                // Avoid rate limiting calculation if log level is disabled
+                WriteIfNotRateLimited(level, exception, messageTemplate, args, sourceLine, sourceFile);
+            }
+        }
+
+        private void WriteIfNotRateLimited(LogEventLevel level, Exception exception, string messageTemplate, object[] args, int sourceLine, string sourceFile)
+        {
+            try
+            {
+                if (_rateLimiter.ShouldLog(sourceFile, sourceLine, out var skipCount))
+                {
+                    var newArgs = new object[args.Length + 1];
+                    Array.Copy(args, newArgs, args.Length);
+                    newArgs[args.Length] = skipCount;
+
+                    _logger.Write(level, exception, messageTemplate + ", {SkipCount} additional messages skipped", newArgs);
+                }
+            }
+            catch
+            {
+                try
+                {
+                    var ex = exception is null ? string.Empty : $"; {exception}";
+                    var properties = args.Length == 0
+                        ? string.Empty
+                        : "; " + string.Join(", ", args.Select(x => x.ToString()));
```

Comment:
No need to call ToString, there's an object[] overload that does it for you
```suggestion
                        : "; " + string.Join(", ", args);
```

Reply:
Didn't know that, cheers!